src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Completion/RazorCompletionEndpointTest.cs

```diff
@@ -616,5 +596,34 @@ private static RazorCodeDocument CreateCodeDocument(string text)
             codeDocument.SetTagHelperContext(tagHelperDocumentContext);
             return codeDocument;
         }
+
+        private class TestLSPDocumentContextFactory : DocumentContextFactory
```

Comment:
Probably it will make sense to make this class available to the other tests?

Reply:
ya good point. This will probably become pretty normal to use. I'll promote